idaes/gas_solid_contactors/properties/oxygen_iron_OC_oxidation/gas_phase_thermo.py

```diff
@@ -0,0 +1,798 @@
+                                 ('CO2', 1): 3.69, ('CO2', 2): -0.3838,
+                                 ('CO2', 3): 964, ('CO2', 4): 1.86e6,
+                                 ('H2O', 1): 6.204e-6, ('H2O', 2): 1.3973,
+                                 ('H2O', 3): 0, ('H2O', 4): 0}
+        self.therm_cond_param = Param(self.component_list,
+                                      range(1, 10),
+                                      mutable=True,
+                                      initialize=therm_cond_param_dict,
+                                      doc="Thermal conductivity constants")
+
+        # Component diffusion volumes:
+        # Ref: (1) Prop gas & liquids (2) Fuller et al. IECR, 58(5), 19, 1966
+        diff_vol_param_dict = {'O2': 16.6, 'N2': 17.9,
+                               'CO2': 26.9, 'H2O': 13.1}
+        self.diff_vol_param = Param(self.component_list,
+                                    mutable=True,
+                                    initialize=diff_vol_param_dict,
+                                    doc="Component diffusion volumes")
+
+        self._eps = Param(initialize=1e-8,
+                          doc="Smooth abs reformulation parameter")
+
+    @classmethod
+    def define_metadata(cls, obj):
+        obj.add_properties({
+                'flow_mol': {'method': None, 'units': 'mol/s'},
+                'pressure': {'method': None, 'units': 'bar'},
+                'temperature': {'method': None, 'units': 'K'},
+                'mole_frac_comp': {'method': None, 'units': None},
+                'mw': {'method': '_mw', 'units': 'kg/mol'},
+                'cp_mol': {'method': '_cp_mol', 'units': 'kJ/mol.K'},
+                'cp_mol_comp': {'method': '_cp_mol_comp',
+                                'units': 'kJ/mol.K'},
+                'cp_mass': {'method': '_cp_mass', 'units': 'kJ/kg.K'},
+                'dens_mol': {'method': '_dens_mol',
+                             'units': 'mol/m^3'},
+                'dens_mol_comp': {'method': '_dens_mol_comp',
+                                  'units': 'mol/m^3'},
+                'dens_mass': {'method': '_dens_mass',
+                              'units': 'kg/m^3'},
+                'enth_mol': {'method': '_enth_mol', 'units': 'kJ/mol'},
+                'enth_mol_comp': {'method': '_enth_mol_comp',
+                                  'units': 'kJ/mol'},
+                'visc_d': {'method': '_visc_d', 'units': 'kg/m.s'},
+                'therm_cond': {'method': '_therm_cond', 'units': 'kJ/m.K.s'},
+                'diffusion_comp': {'method': '_diffusion_comp',
+                                   'units': 'cm2/s'}})
+
+        obj.add_default_units({'time': 's',
+                               'length': 'm',
+                               'mass': 'kg',
+                               'amount': 'mol',
+                               'temperature': 'K',
+                               'energy': 'kJ',
```

Comment:
1. Remove energy and holdup from here - they are no longer supported or needed here.
2. We should start moving towards using Pyomo units here. See comment later.

Reply:
@andrewlee94 This is an aside but how do we add derived units to the `define_metadata` method? For example I would like to add derived units for `power` like `pyunits.kJ * pyunits.s ** -1`. I haven't seen an example in any of the IDAES property packages.

Reply:
You cannot. In order to keep the number of unit conversions to a minimum, we force everything to be based on the 5 base quantities. Otherwise, everything single Expression and Constraint would need a unit conversion in it.

In the sense that this results in some poorly scaled variables in the natural forms, that should be addressed using the scaling tools ,not by defining different sets of units for given quantities.

Reply:
I see... I understand the challenges but I thought we want users to be able to specify and use any units they want.

Reply:
@andrewlee94 do the scaling tools allow us to do variable scaling?

Reply:
1. We are looking at tools to handle the input-output level so that we can report variables in any set of units. However, for the models themselves, we need to keep things to a consistent set (we have already seen that unit conversion can be very expensive).

2. The short answer is yes, the scaling tools can do variable scaling. However, the long answer is a bit more complex. First of all, we have found that constraint scaling alone is often enough, unless you are dealing with variables with very small deviations from bounds. To get variable scaling requires using user-defined scaling factors in IPOPT which disables IPOPT's Jacobian scaling. Thus, we need an extra step (for which we have a tool) to replicate this Jacobian scaling before we send the problem to IPOPT.